internal/source/image_registry.go

```diff
@@ -0,0 +1,206 @@
+package source
+
+import (
+	"archive/tar"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containerd/containerd/archive"
```

Comment:
The focus here is just pull in the code, but maybe we should add an issue to make more use of `containerd` for the future?  Seems like `containerd` has image operations that could simplify TLS and authentication handling, streamline pulling images, and optimize unpacking layers.